nebula-logger/core/main/configuration/classes/LoggerCache.cls

```diff
@@ -0,0 +1,230 @@
+    private static final SessionCache SESSION_CACHE_INSTANCE = new SessionCache();
+    private static final TransactionCache TRANSACTION_CACHE_INSTANCE = new TransactionCache();
+
+    @SuppressWarnings('PMD.ApexDoc')
+    private interface Cacheable {
+        Boolean contains(String key);
+        Object get(String key);
+        void put(String key, Object value);
+        void remove(String key);
+    }
+
+    /**
+     * @description The instance of `OrganizationCache` used for any organization-specific caching.
+     *              When Platform Cache is disabled or not available, the transaction cache is instead used.
+     * @return      The singleton instance of `OrganizationCache`
+     */
+    public static OrganizationCache getOrganizationCache() {
+        return ORGANIZATION_CACHE_INSTANCE;
+    }
+
+    /**
+     * @description The instance of `SessionCache` used for any session-specific caching.
+     *              When Platform Cache is disabled or not available, the transaction cache is instead used.
+     * @return      The singleton instance of `SessionCache`
+     */
+    public static SessionCache getSessionCache() {
+        return SESSION_CACHE_INSTANCE;
+    }
+
+    /**
+     * @description The instance of `TransactionCache` used for any transaction-specific caching.
+     *              Cached data is stored internally in-memory for the duration of the transaction.
+     * @return      The singleton instance of `TransactionCache`
+     */
+    public static TransactionCache getTransactionCache() {
+        return TRANSACTION_CACHE_INSTANCE;
+    }
+
+    /**
+     * @description Manages any organization-specific caching, using org platform cache & transaction cache
+     */
+    public class OrganizationCache implements Cacheable {
+        /**
+         * @description Indicates if the specified key has already been added to the cache
+         * @param  key The `String` key to check for within the transaction cache
+         * @return     The `Boolean` result that indicates if the specified key is contained in the cache
+         */
+        public Boolean contains(String key) {
+            if (LoggerParameter.USE_PLATFORM_CACHE == false || getTransactionCache().contains(key)) {
+                return getTransactionCache().contains(key);
+            } else {
+                return ORGANIZATION_CACHE_PARTITION.contains(key);
+            }
+        }
+
+        /**
+         * @description Returns the cached value for the specified key, or `null` if
+         *              the specified key does not exist in the transaction cache
+         * @param  key  The `String` key to check for within the transaction cache
+         * @return      The cached value, or null if no cached value is found for the specified key
+         */
+        public Object get(String key) {
+            if (LoggerParameter.USE_PLATFORM_CACHE == false || getTransactionCache().contains(key)) {
```

Comment:
it _feels_ like there's still a chance for further consolidation of the `get` checks between the `OrganizationCache` and `SessionCache` cacheables, though the Session one does an additional check - not sure how worth it doing something like that would be though, this is probably fine as is

Reply:
Ooooh, that's a good point, they're very similar, I think I can consolidate it into 1 class. The extra check in the Session cache for `isAvailable()` also exists for Org partitions because they both extend the `Partition` class, and [`Partition.isAvailable()`](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_class_cache_Partition.htm#apex_cache_Partition_isAvailable) seems to always return true for instances of `OrgPartition`, so it doesn't hurt for that extra method check to run in that situation. I'll try to consolidate to 1 class that just has references to `Partition` variables (instead of both `OrgPartition` and `SessionPartition`), I think that'll cut out all of the duplication.

Reply:
I went to the docs to try to find the shared class - remembered it existed, failed to find it, and left 🤦‍♂️. I was going to suggest an always true check for the org one, but that felt like too much. Glad to know they share a class that can be used to consolidate these checks further!

Reply:
Yeah, I honestly didn't expect them to share an interface or super class, this was a really nice surprise 🥳

Reply:
Ok, I finished making this change (and other changes based on your other comments), and I like this much better! [Here's the commit](https://github.com/jongpie/NebulaLogger/commit/7fc2727d9bd62c75deb723da826e30a260e08d45) for these changes, if you're interested is seeing the final result ("final" 😆 )